### ISSUE TITLE
Synchronous operations are being implemented

### ISSUE THREAD

**Issue by amazuerar** [ubergeek42/weechat-android#527]
Dear Developer!

My name is Alejandro Mazuera-Rozo, I am a PhD Student at Universidad de los Andes, and at Università della Svizzera italiana. I am part of a research on the usage of network libraries within Android apps. As result of this we identified some code locations that might have network related problems.

In this case, we address the code locations that are related to a potential issue because of implementing synchronous operations. As you might know, using the `execute()` method on a call object will perform a synchronous request which is not recommended and its usage is discouraged.

In **synchronous operations**, tasks are performed one at a time and only when one is completed, the following is unblocked.

In **asynchronous operations**, you can move to another task before the previous one finishes, allowing you to deal with multiple requests simultaneously.
           
Moreover, by avoiding this approach, it is less likely to perform network operations on the main thread. Android requires you to perform network operations on a thread other than the main UI thread; a `NetworkOnMainThreadException` is thrown otherwise.
       
In order to address this issue we recommend you:

1. To see: https://developer.android.com/training/basics/network-ops/connecting
2. To consider an asynchronous approach.

**Potential Code Location with synchronous operations**

https://github.com/ubergeek42/weechat-android/blob/379f0863e9eef70d83462a5d13e8de932eb785b5/app/src/main/java/com/ubergeek42/WeechatAndroid/upload/Upload.kt#L98

**Comment by oakkitten** [comment]
this operation is not performed on the main thread
